bls/src/main/java/tech/pegasys/teku/bls/impl/blst/BlstSignature.java

```diff
@@ -17,34 +17,32 @@
 
 import java.util.List;
 import java.util.Objects;
-import java.util.Optional;
 import java.util.stream.Collectors;
 import org.apache.tuweni.bytes.Bytes;
 import supranational.blst.BLST_ERROR;
 import supranational.blst.P2;
 import supranational.blst.P2_Affine;
 import supranational.blst.Pairing;
+import tech.pegasys.teku.bls.impl.DeserializeException;
 import tech.pegasys.teku.bls.impl.PublicKey;
 import tech.pegasys.teku.bls.impl.PublicKeyMessagePair;
 import tech.pegasys.teku.bls.impl.Signature;
 
 public class BlstSignature implements Signature {
   private static final int COMPRESSED_SIG_SIZE = 96;
 
-  // The valid infinite signature
-  static final BlstSignature INFINITY = new BlstSignature(new P2_Affine(), true);
+  static final BlstSignature INFINITY = new BlstSignature(new P2_Affine());
 
   public static BlstSignature fromBytes(Bytes compressed) {
-    checkArgument(
-        compressed.size() == COMPRESSED_SIG_SIZE,
-        "Expected " + COMPRESSED_SIG_SIZE + " bytes of input but got %s",
-        compressed.size());
     try {
+      checkArgument(
+          compressed.size() == COMPRESSED_SIG_SIZE,
+          "Expected " + COMPRESSED_SIG_SIZE + " bytes of input but got %s",
+          compressed.size());
       P2_Affine ec2Point = new P2_Affine(compressed.toArrayUnsafe());
-      return new BlstSignature(ec2Point, ec2Point.in_group());
+      return new BlstSignature(ec2Point);
     } catch (Exception e) {
-      // Return an invalid infinite signature if deserialisation fails
-      return new BlstSignature(new P2_Affine(), false);
+      throw new DeserializeException("Signature deserialisation failed: " + e.getMessage());
```

Comment:
```suggestion
      throw new DeserializeException("Signature deserialization failed", e);
```

and add to `DeserializeException`:

```
  public DeserializeException(String message, Throwable cause) {
    super(message, cause);
  }
```

Reply:
Done